scrub_hub_vite/src/components/ConversationListComponent.tsx

```diff
@@ -1,37 +1,47 @@
-import { useEffect, useState } from 'react';
-import { isChatProps } from './ChatTypes';
-import ChatComponent, { ChatProps } from './ChatComponent';
+import { useContext, useState } from 'react';
+import ChatComponent, { ChatProps, MessageProps } from './ChatComponent';
 import styles from './chat.module.css';
 
 import { CiSearch } from "react-icons/ci"
+import { IncomingConversationDetails, User, isIncomingConversationDetailsData } from '../models/chat';
+import { LoginContext, LoginState } from '../loginInfo';
 
 export type ConversationProps = {
-	participants: string[],
-	last_message_time?: Date,
-	last_message?: string,
+	participants: User[],
+	last_message?: MessageProps,
 	id: number,
 }
-function ConversationComponent({ participants, last_message_time, last_message }: ConversationProps) {
-	return <div className='relative border-2 border-black p-3'>
-		<p className='absolute right-1 top-0 text-sm text-gray-900'>{last_message_time?.toLocaleTimeString()}</p> {/* TODO: Format as "2min ago" */}
+function ConversationComponent({ participants, last_message }: ConversationProps) {
+	return <div className='relative border-2 border-black p-3 h-26'>
+		<p className='absolute right-1 top-0 text-sm text-gray-900'>{last_message?.time.toLocaleTimeString()}</p> {/* TODO: Format as "2min ago" */}
 		<div className='flex flex-row'>
 			<div className='profilePicture' /> {/* Profile picture placeholder. TODO: Include indicator for people who are online */}
 			<div className='self-center'>
-				<p className='font-bold text-lg'>{participants.join(', ')}</p>
-				<p>{last_message ?? 'Click to start a conversaiton'}</p>
+				<p className='font-bold text-lg'>{participants.map((p) => p.name).join(', ')}</p>
+				<p className='max-h-6 overflow-hidden'>
+					{last_message ? `${last_message.username}: ${last_message.message}` : 'Click to start a conversaiton'}
+				</p>
 			</div>
 		</div>
 	</div>
 }
 
 export type ConversationListProps = {
 	conversations: ConversationProps[],
-	available_users: { name: string, id: number }[],
+	available_users: User[],
 }
 export default function ConversationListComponent({ conversations, available_users }: ConversationListProps) {
+	const userInfo = useContext(LoginContext);
 	const [activeConversation, setConversation] = useState<ChatProps | undefined>();
+	const [nextTemporaryId, setTempId] = useState(-1);
 
-	async function conversationClick(url_path: string) {
+	if (!userInfo.user || userInfo.loggedIn !== LoginState.IN) {
+		// This shouldn't ever happen. Django would redirect us first.
+		window.location.href = `${location.protocol}//${location.host}/authenticate/login`;
+		throw 'Not logged in';
+	}
+
+	async function conversationClick(url_path: string): Promise<ChatProps> {
```

Comment:
First of all. Why is this here, and another place as well. Second of all, why is this functionality NOT separated out into its own function? :)

Reply:
It's there mainly so TypeScript will know that userInfo.user is not undefined. It should be in its own function.

Reply:
Can userInfo.user be undefined though? And is there instead a way where we could just ensure that userInfo.user just cannot ever be undefined? We are after all working in typescript and we should, at least in theory, be able to do that? I think that would be better, but we can just start by at least, creating a function for this.

Reply:
It will be undefined if the user isn't logged in (or, the info isn't present in the browsers local storage). After all, if there is no logged in user there is no user.id or any name associated with the user. Unless we wanted to have guest users? But right now we don't.
But when it is moved to a function, that function can return a LoggedInUser object that is never undefined.

Reply:
But like, if you aren't logged in, can you ever reach this component??

Reply:
just something to think about

Reply:
According to Django, no. As the comment says, Django would redirect us first. But it is technically possible for userInfo.user to be undefined, if the user has for some reason deleted things from local storage.